qa-tests-backend/scripts/run-compatibility.sh

```diff
@@ -119,4 +124,34 @@ shorten_tag() {
     fi
 }
 
+verify_scannerV2_deployed() {
+    local namespace=${1:-stackrox}
+    info "Waiting for Scanner V2 deployment to appear in namespace ${namespace}..."
+    wait_for_object_to_appear "$namespace" deploy/scanner-db 600
+    wait_for_object_to_appear "$namespace" deploy/scanner 300
+    info "** Scanner V2 is deployed in namespace ${namespace}"
+}
+
+verify_scannerV4_deployed() {
+    local namespace=${1:-stackrox}
+    verify_scannerV4_indexer_deployed "$namespace"
+    verify_scannerV4_matcher_deployed "$namespace"
+}
+
+verify_deployment_scannerV4_env_var_set() {
+    local namespace=${1:-stackrox}
+    local deployment=${2:-central}
+    local deployment_env_vars
+    local scanner_v4_value
+
+    deployment_env_vars="$("${ORCH_CMD}" </dev/null -n "${namespace}" get deploy/"${deployment}" -o jsonpath="{.spec.template.spec.containers[?(@.name=='${deployment}')].env}")"
```

Comment:
I am afraid this may not be enough. This will get a deployment, yes, but this does not check if pod is ready. We need to make sure that this check fails if the pod is crashing, starting, image-pull-error - i.e. all states different than `ready` must be rejected.

I support Matthias'es idea about readiness-checks. There are also alternative ways of making sure that scanners are operational.

Reply:
Everything considered it appears to make the most sense to have a readyness-check within the `CertExpiryTest`, which would also reduce the overhead of waiting for scanner to be ready since other tests could already run while Scanner is starting up.

The only problem is that the test is written in groovy, but I'll try to move the checks there.

Reply:
I think, it is fine to leave it where it is. In the future we may have more v4 tests and we would need to move that availability check again.
I would suggest to implement it where it is easier for now as long as it does it's job.

Reply:
I think I found a good compromise here. I wrote a bash script that checks whether the scanner pod is available and scanner is ready, and now I'm calling this script from within `CertExpiryTest`.

Since the script does some waiting for scanner to be ready it saves time to run it later (when CertExpiryTest comes up) rather than earlier (when setting up compatibility tests), but it's also easy enough to move this call should we need to later.